packages/core/src/utils/universalFetch.ts

```diff
@@ -0,0 +1,28 @@
+export async function setupFetch() {
+  if (typeof window !== "undefined" && window.fetch) {
+    return window.fetch;
+  }
+
+  if (typeof process !== "undefined" && process.versions.node) {
+    const [major] = process.versions.node.split(".").map(Number);
+
+    if (major < 18) {
+      const mod = await import("node-fetch");
+      return mod.default as unknown as typeof fetch;
+    }
+
+    return globalThis.fetch;
+  }
+
+  return undefined;
+}
+
+export async function universalFetch(
```

Comment:
I'm not sure what to name this function. If you have any good ideas, please recommend.
@manudeli